### PR TITLE
Sync: Preserve the home_url, site_url and network_url

### PR DESCRIPTION
**Pull request by enejb**
This PR tries to make the urls more stable in the scheme sense. Since
the scheme can change depending on what url the user uses to send the
data from on shutdown.

This PR tries to fix this by matching the schema of the url more
closely to the url option defined in the general settings of the site.

-------------------
- [ ] Create **[unit tests](https://github.com/Automattic/jetpack/tree/master/tests)** if you can. If you're not familiar with Unit Testing, you can check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/).

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-callables.php [hunk 1]
```diff
@@ -123,4 +122,43 @@ function test_white_listed_callables_doesnt_get_synced_twice() {
...
+		$_SERVER['HTTPS'] = 'on';
+		delete_transient( Jetpack_Sync_Client::CALLABLES_AWAIT_TRANSIENT_NAME );
+		$this->client->do_sync();
+
+		unset( $_SERVER['HTTPS'] );
+		$this->assertEquals( $home_url, $this->server_replica_storage->get_callable( 'home_url' ) );
+	}
+
+	function test_preserve_schema() {
+		update_option( 'banana', 'http://example.com' );
```
Comment (lezama) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: 🍌

**Comment by lezama** [comment]
I am not sure what's the best solution here, what about sending the three callables protocol agnostic, and also send the options aside?

cc @gravityrail

**Comment by enejb** [comment]
Sending protocol agnostic urls and the options don't really help us at the end of the day we want to have a url that we can use. I think having one that we are confident in (that won't change is the best approach) Also I think the problem is not really something that affects to many sites.

File: tests/php/sync/test_class.jetpack-sync-callables.php [hunk 2]
```diff
@@ -123,4 +122,43 @@ function test_white_listed_callables_doesnt_get_synced_twice() {
...
+		// this sets is_ssl() to return true.
+		$_SERVER['HTTPS'] = 'on';
+		delete_transient( Jetpack_Sync_Client::CALLABLES_AWAIT_TRANSIENT_NAME );
+		$this->client->do_sync();
+
+		unset( $_SERVER['HTTPS'] );
+		$this->assertEquals( $home_url, $this->server_replica_storage->get_callable( 'home_url' ) );
+	}
+
+	function test_preserve_schema() {
```
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: Typo, schema -> scheme

**Comment by gravityrail** [comment]
lgtm, I like this approach 👍

**Pull request merged**